src/main/java/org/codehaus/mojo/license/LicenseMojoUtils.java

```diff
@@ -134,7 +122,7 @@ private static URL prepareUrl( final URL resolvedUrl, final File deprecatedFile,
         {
             try
             {
-                return defaultPath.toUri().toURL();
+                return defaultPath.toUri().toURL().toString();
```

Comment:
`deprecatedFile.toURI().toString()`. `.toURL()` is redundant there.

Reply:
For what I can read in the javadoc of `Path.toUri()`  There is no guarantee that the URI may be used to construct a java.io.File that means the I could got an Uri that could not be be handled by standard URL handlers. Moreover the `toUri()` does not automatically escape characters that are illegal in URLs.

This code was already in place and I would not change. With handler approach (maybe think different than I implemented) these changes would not be necessary.

Reply:
> in the javadoc of `Path.toUri()` There is no guarantee that the URI may be used to construct a java.io.File

Do we ever directly create a `java.io.File` out of the resulting URI string? I am not aware of any such execution path. 

What we do is using the string returned by `LicenseMojoUtils.prepareUrl()` to construct a URL in `UrlRequester.getFromUrl()` and I think that should work well with `defaultPath.toUri().toString()`. 

The JavaDoc in `Path.toUri()` says that something like `new File(new URI(stringPath))` may not work for UNC stringPaths. But as I said, our path of execution is different - it's something like `new URL(new URL(path.toUri().toString())).openStream()`. 

Actually the best way to figure out would be to write a simple test checking if `path.toUri().toString()` and `path.toUri().toURL().toString()` are equal for UNC paths. If they are, we can safely remove `.toURL()`. Let me try that.

Reply:
Here is the test: https://github.com/mojohaus/license-maven-plugin/pull/263

Reply:
The test is passing on Windows and fails on Linux (which is kinda expected because UNC paths are not supposed to work without being mounted). So I think the test proves that `.toURL()` can be removed. WDYT?